packages/math-input/src/components/keypad/navigation-button.tsx

```diff
@@ -0,0 +1,142 @@
+import Clickable from "@khanacademy/wonder-blocks-clickable";
+import Color from "@khanacademy/wonder-blocks-color";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+import ButtonAsset from "./button-assets";
+
+import type Key from "../../data/keys";
+import type {KeyConfig, ClickKeyCallback} from "../../types";
+
+export type KeypadButtonProps = {
+    keyConfig: KeyConfig;
+    onClickKey: ClickKeyCallback;
+};
+
+function getCoords(key: Key) {
+    switch (key) {
+        case "UP":
+            return [1, 0];
+        case "RIGHT":
+            return [2, 1];
+        case "DOWN":
+            return [1, 2];
+        case "LEFT":
+            return [0, 1];
+        default:
+            throw new Error(`Invalid key: ${key}`);
+    }
+}
+
+function getStyles(key: Key) {
+    switch (key) {
+        case "UP":
+            return styles.up;
+        case "RIGHT":
+            return styles.right;
+        case "DOWN":
+            return styles.down;
+        case "LEFT":
+            return styles.left;
+        default:
+            throw new Error(`Invalid key: ${key}`);
+    }
+}
+
+export default function NavigationButton({
+    keyConfig,
+    onClickKey,
+}: KeypadButtonProps) {
+    const key = keyConfig.id;
+    const coord = getCoords(key);
+    const directionalStyles = getStyles(key);
+
+    return (
+        <View
+            style={{
+                gridColumn: coord[0] + 1,
+                gridRow: coord[1] + 1,
+            }}
+        >
+            <Clickable
+                onClick={(e) => onClickKey(keyConfig.id, e)}
+                style={styles.clickable}
+                aria-label={keyConfig.ariaLabel}
+            >
+                {({hovered, focused, pressed}) => {
+                    return (
+                        <View style={styles.outerBoxBase}>
+                            <View
+                                style={[
+                                    styles.base,
+                                    directionalStyles,
+                                    hovered && styles.hovered,
+                                    focused && styles.focused,
+                                    pressed && styles.pressed,
+                                ]}
+                            >
+                                <ButtonAsset id={keyConfig.id} />
+                            </View>
+                        </View>
+                    );
+                }}
```

Comment:
This is extremely minor, but for these functions that just return a React node, it's possible to collapse it slightly to this. The one nice thing is it unindents the JSX by one level.

```suggestion
                {({hovered, focused, pressed}) => (
                    <View style={styles.outerBoxBase}>
                        <View
                            style={[
                                styles.base,
                                directionalStyles,
                                hovered && styles.hovered,
                                focused && styles.focused,
                                pressed && styles.pressed,
                            ]}
                        >
                            <ButtonAsset id={keyConfig.id} />
                        </View>
                    </View>
                )}
```

Reply:
I kinda don't like `() => ()` as much as `() => { return () }` because I abuse `console.log` when working so I need a place to put logs. But I think you're still right so I changed it.